### PR TITLE
Allow multiline text

### PR DESCRIPTION
**Pull request by jillesme**
# Multiline text!

Based on my old (closed) PR https://github.com/excalidraw/excalidraw/pull/385/

### CODE REVIEW

File: src/renderer/renderElement.ts [hunk 1]
```diff
@@ -144,11 +144,13 @@ export function renderElement(
...
-    context.fillText(
-      element.text,
-      0,
-      element.baseline || element.actualBoundingBoxAscent || 0
-    );
+    // Canvas does not support multiline text by default
+    const lines = element.text.replace(/\r\n?/g, "\n").split("\n");
+    const lineHeight = element.height / lines.length;
+    const offset = element.height - element.baseline;
+    for (let i = 0; i < lines.length; i++) {
```
Comment (lipis) [inline src/renderer/renderElement.ts]: Use `index`

Comment (dwelle) [reply src/renderer/renderElement.ts]: @lipis I noticed you focus too much on unimportant minutiae that only adds unnecessary friction.

Comment (lipis) [reply src/renderer/renderElement.ts]: Having a nice a consistent backend helps a lot in a long run:  https://eslint.org/docs/rules/id-length

Comment (dwelle) [reply src/renderer/renderElement.ts]: Looking up `for (` in our codebase returns `i` 2x, `pos` 2x. So to be consistent, sticking to `i` is the right move here.

Nevertheless, this is still a very minor thing -- let's focus on bigger things. (And I'm also not referring just to this case.)

Comment (lipis) [reply src/renderer/renderElement.ts]: It's not a bad thing too change them all wherever possible

**Comment by jillesme** [comment]
@vjeux I know I have merge powers but I'll only merge with your approval ✅

File: src/renderer/renderElement.ts (hunk 1 above)
Comment (jillesme) [reply src/renderer/renderElement.ts]: @lipis changing the convention `i` to `index` adds very little value. You mention the id-length rule for ESLint but this does not apply for loops, where `i` and `j` are the convention. See example https://eslint.org/docs/rules/for-direction#top

I really appreciate that you take the time to review my code! However, I'd have to agree with @dwelle that those unimportant opinionated nits add unnecessary friction.

**Comment by lipis** [comment]
Have you try exporting it and reloading with the multiline text?
(get shareable URL)

**Comment by jillesme** [comment]
@lipis yes but that's not going to work on the preview domain due to CORS. I exported it as JSON and imported it again and it works fine.

**Comment by dwelle** [comment]
LGTM. In the future we'll probably want to make the text centered, but that will take more work so let's leave it for future PRs.

**Pull request merged**